### PR TITLE
Fix/ft 62/code refactoring

### PR DESCRIPTION
**Pull request by DenysLutskiy**

### CODE REVIEW

File: src/doctors/doctors.module.ts [hunk 1]
```diff
@@ -1,26 +1,21 @@
...
-import { UsersService } from 'src/users/users.service';
-import { SpecializationsService } from 'src/specializations/specializations.service';
-import { AuthService } from 'src/auth/auth.service';
-import * as REDIS_CONFIG from 'src/config/redis';
+import { AuthModule } from 'src/auth/auth.module';
+import { SpecializationsModule } from 'src/specializations/specializations.module';
 
 @Module({
   imports: [
+    AuthModule,
```
Comment (denismukhopadov) [inline src/doctors/doctors.module.ts]: why do you need auth module everywhere ? bc of guard ?

File: src/doctors/doctors.resolver.ts [hunk 2]
```diff
@@ -35,11 +36,13 @@ export class DoctorsResolver {
...
+    return await this.usersService.findOneById(doctor.user.id);
   }
 
   @ResolveField('specialization')
   async getSpecialization(@Parent() doctor: Doctor) {
-    return await this.specializationsService.findOne(doctor.specialization.id);
+    return await this.specializationsService.findOneById(
+      doctor.specialization.id,
+    );
   }
```
Comment (d-smyrnov) [inline src/doctors/doctors.resolver.ts]: It's strange that to query specialization it should already be in doctor. In that case, you can just return it. Or you can use a doctor.specializationId field.

File: src/doctors/doctors.service.ts [hunk 3]
```diff
@@ -1,33 +1,27 @@
...
-      id: createDoctorInput.userId,
-    });
-    console.log(spec);
-
+    const spec = await this.specializationsService.findOneById(
+      createDoctorInput.specializationId,
+    );
+    const user = await this.usersService.findOneById(createDoctorInput.userId);
     doctor.specialization = spec;
     doctor.user = user;
```
Comment (d-smyrnov) [inline src/doctors/doctors.service.ts]: You can just set the id columns to create a relation. (Not necessary to query the entities spec and user, as you don't check for existence or so ever).

File: src/users/users.service.ts [hunk 4]
```diff
@@ -73,8 +73,13 @@ export class UsersService {
...
-  async findOne(id: string) {
+  async findOneById(id: string) {
     const user = await this.usersRepository.findOne(id);
     return user;
   }
+
+  async findOneWithOptions(options: object) {
+    const user = await this.usersRepository.findOne(options);
+    return user;
+  }
```
Comment (d-smyrnov) [inline src/users/users.service.ts]: No return types and variable definitions could be omitted.

**Pull request closed without merging**